packages/core/crates/core-network/src/ping.rs

```diff
@@ -1,7 +1,10 @@
-use std::time::Duration;
+use std::{time::Duration, pin::Pin, result};
 
+use async_trait::async_trait;
 use futures::{
+    channel::mpsc,
```

Comment:
How is it different if we used use `async_std::channel` since we have `async_std` already in?

Reply:
I'd opt to using as much generics as possible, instead of the async-std lock-in. In the end it can be reasonable to assume tokio might be more suitable once everything is migrated.